src/operator/tensor/ordering_op-inl.h

```diff
@@ -401,8 +401,6 @@ void TopKImpl(const RunContext &ctx,
     mxnet::op::SortByKeyWorkspaceSize<int, int, xpu>(src.Size()));
   temp_size = std::max(temp_size,
     mxnet::op::SortByKeyWorkspaceSize<int, DType, xpu>(src.Size()));
-  temp_size = std::max(temp_size,
-    mxnet::op::SortByKeyWorkspaceSize<DType, int, xpu>(src.Size()));
```

Comment:
This line should be useful because we need to do `mxnet::op::SortByKey(dat, ind, is_ascend, &sort_work);`.

Reply:
can't it can be done even now? this change passed all the unit tests, btw.

Reply:
Removing this line may break some corner cases uncovered by the existing test. Since we need to call `SortByKey<DType, int>(dat, ind, ...)`, `SortByKey<int, DType>(batch_id, dat, ...)` and `SortByKey<int, int>(batch_id, ind, ...)`, we should make sure that the temporary storage has enough size for all cases.

Reply:
> may break some corner cases uncovered by the existing test

it does not break any existing tests. I checked that locally. And the tests passed the CI too.

> we should make sure that the temporary storage has enough size for all cases.

I thought about this when making the change. Is there a use-case/unit test you could point to which would break with this change.